revscoring/scorers/svc.py

```diff
@@ -51,52 +54,73 @@ def train(self, values_scores, balanced_weight=True):
             'seconds_elapsed': time.time() - start
         }
     
-    def score(self, values, probabilities=False):
+    def score(self, values):
         """
         :Returns:
             An iterable of dictionaries with the fields:
             
             * predicion -- The most likely class
-            * probabilities -- (optional) A vector of probabilities
-                               corresponding to the classes the classifier was
-                               trained on.  Generating this probability is
-                               slower than a simple prediction.
+            * probability -- A mapping of probabilities for input classes
+                             corresponding to the classes the classifier was
+                             trained on.  Generating this probability is
+                             slower than a simple prediction.
         """
         scaled_values = list(self._scale_and_center(values, self.feature_stats))
-        if not probabilities:
-            for prediction in self.svc.predict(scaled_values):
-                yield {'prediction': prediction}
-        else:
-            for pred, probas in zip(self.svc.predict(scaled_values),
-                                   self.svc.predict_proba(scaled_values)):
-                yield {'prediction': pred,
-                       'probabilities': list(probas)
-                }
+        predictions = self.svc.predict(scaled_values)
+        probabilities = (
+            {c:proba for c, proba in zip(self.svc.classes_, probas)}
+            for probas in self.svc.predict_proba(scaled_values)
+        )
+        
+        for prediction, probability in zip(predictions, probabilities):
+            yield {
+                'prediction': prediction,
+                'probability': probability
+            }
                 
         
     
-    def test(self, values_scores):
+    def test(self, values_labels, comparison_class="auto"):
         """
         :Returns:
             A dictionary of test statistics with the fields:
             
             * mean.accuracy -- The mean accuracy of classification
+            * auc --
+            * table --
+            * roc
+                * fpr --
+                * tpr --
+                
         """
-        values, scores = zip(*values_scores)
+        values, labels = zip(*values_labels)
+        values, labels = list(values), list(labels)
+        
+        scores = list(self.score(values))
+        
+        if comparison_class == "auto":
+            comparison_class = self.svc.classes_[1]
+        elif comparison_class not in self.svc.classes_[1]:
```

Comment:
You sure about these two lines above? It seems `self.svc.classes_[1]` is either a class (so you can assign it to `comparison_class`) or composed of many classes (so that you can check if `comparison_class` is `in` it). Which one is correct? Or did I miss something?